nntrainer/layers/layer_context.h

```diff
@@ -235,7 +235,7 @@ class InitLayerContext {
   static VarGradSpecV2
   outSpec(const TensorDim &dim, const std::string &name = "out",
           TensorLifespan ls = TensorLifespan::FORWARD_FUNC_LIFESPAN,
-          TensorLifespan grad_ls = TensorLifespan::BACKWARD_FUNC_LIFESPAN);
+          TensorLifespan grad_ls = TensorLifespan::CALC_GRAD_DERIV_LIFESPAN);
```

Comment:
`BACKWARD_FUNC_LIFESPAN` contains `apply gradient` step.
IMHO, Isn't it better to remain it as default while `calc grad` is set?

Reply:
In that case, we will have unnecessary execution order for the input tensor, which is not desirable.  We do not need the outgoing gradient to compute the apply gradient.

Reply:
OK. thanks fo checking.